### ISSUE TITLE
Quantized binary with high execution time

### ISSUE THREAD

**Issue by krishnak** [kendryte/nncase#79]
I tried to recreate from scratch the RFB-320.kmodel that you have in the nncase repository.

I used the RFB-320.caffemodel from ultra fast face detect repository and did the following
`./ncc compile RFB-320.caffemodel test.kmodel -i caffe -o kmodel -t k210 --dataset images`

A kmodel file was produced without any errors.

On the K210 inference works but it is having a high inference time, similar to the ultra_face.kmodel which takes around 1sec per frame.

There is something significantly different between your RFB-320.kmodel creation and the command above.

Because your RFB-320.kmodel takes only 180ms for inference.

**There are few quantized binary layers having a high execution time in the model produced by me. It is there in the ultra_face.kmodel as well.**

What needs to be done to get a sleek and fast model like your RFB-320.kmodel.

My images folder has all the images from ulffd imgs folder. The model size and performance remains the same whether one image or multiple images are present during nncase compile. i.e my model takes 1000ms to infer on the kendryte k210 which is similar to the ultra_face.kmodel performance

There is some step that is missing which gives a nicer model like yours.

@sunnycase

**Comment by sunnycase** [comment]
You can use the nncase in the CI which support accelerate dilated conv2d using KPU (you should also update to the newest standalone SDK).

**Comment by krishnak** [comment]
I thought so, I downloaded the source and trying to build 
 Conan: Found program /usr/local/bin/conan
-- Conan: Version found Conan version 1.21.1

However, I get an error

/home/krishnak/visualstudioworkspace/nncase-master/src/ir/llir/../include/llir/ir_types.h:17:10: fatal error: xtensor/xshape.hpp: No such file or directory
 #include <xtensor/xshape.hpp>
          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.

I am not sure why xtensor dependency didn't get installed.

**Comment by krishnak** [comment]
@sunnycase 

refer to earlier message, when I try to install Xtensor I get this error,  please advise.

nncase-master/out$ conan install xtensor/0.20.10@omaralvarez/public-conan
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

xtensor/0.20.10@omaralvarez/public-conan: Not found in local cache, looking in remotes...
xtensor/0.20.10@omaralvarez/public-conan: Trying with 'sunnycase'...
xtensor/0.20.10@omaralvarez/public-conan: Trying with 'conan-center'...
xtensor/0.20.10@omaralvarez/public-conan: Trying with 'bincrafts'...
ERROR: Unable to find 'xtensor/0.20.10@omaralvarez/public-conan' in remotes

**Comment by minux** [comment]
have you updated git submodules when cloning the repository?
xtensor should be included as a submodule of nncase.

**Comment by krishnak** [comment]
thank you for your fast reply @minux  @sunnycase - much appreciated.

You were right, those directories were empty. After cloning the third party directory, everything builds properly.

I still have one minor issue with this. I have created a kmodel using 

./ncc compile  ~/nncase/RFB-320.caffemodel test.kmodel -i caffe --dataset ~/nncase/images/

Both your RFB-320.kmodel on github and the output of the above command are of exactly same file size.

When I execute both the models, I get an execution time of 190ms, there is a difference of few microseconds between the two models.

However when I run inference on a particular image with
    ultra_face_init(320, 240, 0.3, 0.05, -1);

Your model is detecting 7 faces.
My model is detecting only 5 faces.

The ground truth is 7 faces.
this image is not in the training dataset. I included it in the image dataset during the creation of my kmodel file

However when I use  a different image which is part of the original ultra fast face detector, your model and mine produce exactly the same timing and detects equal number of faces.

I will try few more distinct images to see whether this can be reproduceable across images and open a separate issue if it can be reproduced.

Many thanks once again.
